inference-engine/src/transformations/src/transformations/common_optimizations/nearest_neighbor_umpsampling_fusion.cpp

```diff
@@ -0,0 +1,195 @@
+        if (!current_input) return {};
+
+        const auto current_input_axis = std::dynamic_pointer_cast<opset8::Constant>(current_input->input_value(1).get_node_shared_ptr());
+        if (!current_input_axis || current_input_axis->cast_vector<int64_t>() != std::vector<int64_t>{0}) return {};
+
+        const auto unsqueezed_const = std::dynamic_pointer_cast<opset8::Constant>(current_input->input_value(0).get_node_shared_ptr());
+        if (!unsqueezed_const) return {};
+
+        const auto unsqueezed_const_value = unsqueezed_const->cast_vector<int64_t>();
+        if (unsqueezed_const_value.size() != 1) return {};
+
+        input_constants[i - 1] = unsqueezed_const_value[0];
+    }
+
+    if (input_constants.back() != static_cast<int64_t>(input_shape.back())) return {};
+
+    input_constants.pop_back();
+
+    return input_constants;
+}
+} // namespace
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::NearestNeighborUpsamplingFusion, "NearestNeighborUpsamplingFusion", 0);
+
+ngraph::pass::NearestNeighborUpsamplingFusion::NearestNeighborUpsamplingFusion() {
+    MATCHER_SCOPE(NearestNeighborUpsamplingFusion);
+    auto input = ngraph::pattern::any_input();
+    auto concat_1 = pattern::wrap_type<opset8::Concat>();
+    auto concat_2 = pattern::wrap_type<opset8::Concat>();
+    auto reshape_1 = pattern::wrap_type<opset8::Reshape>({input, concat_1});
+    auto mul_const = pattern::wrap_type<opset8::Constant>();
+    auto mul = pattern::wrap_type<opset8::Multiply>({reshape_1, mul_const});
+    auto reshape_2 = pattern::wrap_type<opset8::Reshape>({mul, concat_2});
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        auto &pattern_to_output = m.get_pattern_value_map();
+
+        const auto reshape_2_node = std::dynamic_pointer_cast<opset8::Reshape>(pattern_to_output.at(reshape_2).get_node_shared_ptr());
+        const auto mul_node = std::dynamic_pointer_cast<opset8::Multiply>(pattern_to_output.at(mul).get_node_shared_ptr());
+
+        if (!reshape_2_node || !mul_node || mul_node->get_input_partial_shape(1).is_dynamic()) return false;
+
+        const auto mul_const_node = std::dynamic_pointer_cast<opset8::Constant>(mul_node->input_value(1).get_node_shared_ptr());
+        if (!mul_const_node) return false;
+
+        const auto reshape_1_node = std::dynamic_pointer_cast<opset8::Reshape>(pattern_to_output.at(reshape_1).get_node_shared_ptr());
+        if (!reshape_1_node || reshape_1_node->get_input_partial_shape(0).is_dynamic()) return false;
+
+        uint64_t input_rank = static_cast<uint64_t>(reshape_1_node->get_input_partial_shape(0).rank().get_length());
+        const auto mul_const_shape = mul_const_node->get_output_shape(0);
+        const auto scales = get_scales_from_mul_const_shape(mul_const_shape, input_rank);
+        if (scales.empty() || std::all_of(scales.begin(), scales.end(), [](float s) { return s == 1.0f;})) { return false; }
```

Comment:
magic const

Reply:
Where is a magic? If we have all scales `1.0f`, this means that `Interpolate` doesn't need.